### PR TITLE
Implement +,-,/, * and negate operators for Color type

### PR DESCRIPTION
**Pull request by bncastle**
Implement +,-,/, * and negate operators for Color type.
~~Note: Currently, none of the operations affect the alpha value of the
color.~~
This pull request addresses issue #11129

### CODE REVIEW

**Comment by Noshyaar** [comment]
Thanks for your contribution.
1. Please remove commented-out codes.
2. Please clang-format changed files.
3. Squash commits together. ([PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request))

File: core/color.cpp [hunk 1]
```diff
@@ -400,3 +400,121 @@ Color::operator String() const {
...
+		p_color.r == 0 ? 1 : CLAMP(r / p_color.r, 0.0, 1.0),
+		p_color.g == 0 ? 1 : CLAMP(g / p_color.g, 0.0, 1.0),
+		p_color.b == 0 ? 1 : CLAMP(b / p_color.b, 0.0, 1.0),
+		a);
+		// p_color.a == 0 ? 1 : CLAMP(a / p_color.a, 0.0, 1.0));
+}
+
+Color Color::operator/(const real_t &rvalue) const {
+	
+		if(rvalue == 0) return Color(1,1,1,1);
```
Comment (Noshyaar) [inline core/color.cpp]: `Color(1,1,1,1)` might cause issue if #11074 gets merged.

Comment (bncastle) [reply core/color.cpp]: Ok. I'll fix it.

**Comment by akien-mga** [comment]
Looks fine, but what's the consensus about the alpha channel? Shouldn't we add it too?
With the current implementation, the operators are not commutative (`color1 + color2 != color2 + color1`), which could lead to some weird bugs :)

**Comment by bncastle** [comment]
I was unsure if including the alpha in these operations was a good idea, but your example would be an issue. I'm happy to add the alpha channels in the calculations if no one else has any objections.

**Comment by bncastle** [comment]
The Color operators included in this pull request now affect the alpha channel as well. This addresses the issue raised by @akien-mga

**Comment by akien-mga** [comment]
It doesn't build with GCC and Clang: https://travis-ci.org/godotengine/godot/builds/275078525?utm_source=github_status&utm_medium=notification

**Comment by bncastle** [comment]
It was a copy/paste error on my part. Sorry about that. It's fixed now.

**Comment by vnen** [comment]
I'm late, but I find this a bit confusing. Why are the values clamped to the 0-1 range? AFAIK colors can have values greater than one (to simulate overexposure). I'm not sure if they can be negative though, so I guess it's okay to limit at zero.

**Comment by bncastle** [comment]
@vnen I didn't realize that, but you're right. I played around with setting rgb values to values > 1 i.e. ```Color(1.5,1.5,1.5,1)``` and it worked fine. Leaving in the 0 clamping would probably be a good idea although not strictly necessary in my tests (I was able to invert an "overexposed" color and it turned black). I'd like to submit a patch to remove the max value clamping if @akien-mga  and /or @Noshyaar agree that this is the right thing to do.

**Pull request merged**